src/backend/base/langflow/memory.py

```diff
@@ -164,7 +164,9 @@ async def aupdate_messages(messages: Message | list[Message]) -> list[Message]:
                 # Convert flow_id to UUID if it's a string preventing error when saving to database
                 if msg.flow_id and isinstance(msg.flow_id, str):
                     msg.flow_id = UUID(msg.flow_id)
-                session.add(msg)
+                result = session.add(msg)
+                if asyncio.iscoroutine(result):
```

Comment:
Just a comment here: @HzaRashid NoopSession is class created by us. In this case, the NoopSession class is incorrect. The `add` method is `sync` and we implemented it as `async` so the correct fix would be to change the NoopSession class.